mantidproject/mantid
# GEM texture mode step 3: save MAUD calibration file and reverse bank numbers

A new algorithm, `SaveGEMMAUDParamFile` was implemented. This algorithm reads various calibration parameters needed by MAUD from a focused workspace and a GSAS calibration file, and outputs them to a MAUD-readable format.

The new algorithm is incorporated into the `isis_powder` routines for GEM.

**To test:**

As always, if not reviewing from ISIS, you'll need to change the second line and do some setup locally - ping me a message for an explanation of how to set up ISIS_Powder directories, as it can be fiddly.

The `.maud` format has been validated by GEM scientists, so just make sure the following runs OK, and the ``.maud`` file is saved in ``\\olympic\Bablyon5\Public\JoeRamsay\GEM\Output\cycle_18_1\GEM``

```
from isis_powder import Gem

gem = Gem(config_file=r"\\olympic\Babylon5\Public\JoeRamsay\GEM\Gem_config_example.yaml",
          vanadium_normalisation=False)

gem.focus(run_number=84734,
          texture_mode=True)
```

Try setting `save_maud_calib` to ``False`` in the `gem.focus` line - the file shouldn't be saved. Have a go a breaking it and code review.

Fixes #22526 

Does this update require release notes?

[Here](https://github.com/mantidproject/mantid/commit/bd6fbba6b7b709f200e08eba464f3c4bed67d272)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
